### ISSUE TITLE
Kernel Not Initializing in JupyterLab

### ISSUE THREAD

**Issue by hakasapl** [WolframResearch/WolframLanguageForJupyter#88]
Hello,

I recently attempted to install this kernelspec into a centralized JupyterLab/JupyterHub installation for a research computing cluster. As such, we use environment modules, and our instance of mathematica is an environment module. This was my process for doing so:
1. I ran `./configure-jupyter.wls add` with no errors. I made sure to clone the repo and run that script in a centralized location that all nodes could access, because I saw that the kernelspec refers back to this repo. In my case thats an NFS mount.
1. I copied the generated kernelspec folder from my user directory to the location of the jupyterlab install where my other kernels are stored. I checked the kernelspec json file to verify that all the files it was looking for can be accessed anywhere on the cluster.
1. Mathematica shows up in jupyterlab, but upon opening a notebook, the kernel is stuck "initializing" for half a minute and then fails.

I don't have any more debug information because I'm not sure where to find it. Are there runtime logs for kernelspecs in jupyterlab? All the GUI gives me is the "no kernel" error. Please let me know if I can provide more info.

**Comment by cc-wr** [comment]
I am working to reproduce this issue.

**Comment by ecbrown** [comment]
Hello, I encountered this issue with a headless Ubuntu Server.  When I installed (with tasksel) the Ubuntu Desktop, I was able to start the kernel.  Also, I installed Wolfram Player, because that was the context of a libGL.so error which I believe was the problem solved by upgrading to full Desktop.

**Comment by okofish** [comment]
To @ecbrown's `libGL.so` issue - I install the following packages in order to get the Wolfram Engine (including child frontends) to work in a headless Ubuntu Server environment:
```shell
apt-get install -y xz-utils avahi-daemon ca-certificates netbase libgl1-mesa-glx libfontconfig1 libasound2
```
If I recall correctly, the last three packages (`libgl1-mesa-glx libfontconfig1 libasound2`) are dependencies of the frontend.

**Comment by aaszodi** [comment]
I have exactly the same problem as OP. I run a headless Ubuntu 20.04 server. `okofish`-s suggestion to install additional GL libraries did not help.

**Comment by GX-Swjtu** [comment]
> 为了@ecbrown的问题 - 我安装以下包，以便让 Wolfram 发动机（包括儿童前端）在无头 Ubuntu 服务器环境中工作：`libGL.so`
> 
> ```shell
> apt-get install -y xz-utils avahi-daemon ca-certificates netbase libgl1-mesa-glx libfontconfig1 libasound2
> ```
> 
> 如果我记错的话，最后三个包（）是前端的依赖。`libgl1-mesa-glx libfontconfig1 libasound2`

Successful operation, thank you

**Comment by AnonymoZ** [comment]
Yeah, the Kernel sometimes becomes rotten and needs be replaced. It happens occasionally, and as things stand I cannot predict when it will happen next. It is like a radioactive decay.

Whatever, just re-install the Kernel as if you just downloaded it for your new computer.
Even keep a ‘.bat’ file that does it for you each time it happens. Not a big deal. 
Here, write something like this and rename the folder as ‘**Repair Corrupted Kernel.bat**’. Keep it in your utils folder:
```
cd (Path to Kernel)
wolframscript -file configure-jupyter.wls add
```
